src/test/java/edu/hm/hafner/analysis/ReportPrinterTest.java

```diff
@@ -52,6 +57,248 @@ void shouldPrintAllIssuesToPrintStream() {
+        // assert
+        for (final Issue issue : report) {
+            verify(logger).error(issue.toString());
+            verify(logger, never()).warn(issue.toString());
+            verify(logger, never()).info(issue.toString());
+            verify(logger, never()).trace(issue.toString());
+        }
+    }
+
+    @Test
+    void shouldSLF4JLogWarningHigh() {
+        // arrange
+        final Report report = new Report();
+        report.add(
+                new IssueBuilder()
+                        .setSeverity(Severity.WARNING_HIGH)
+                        .setMessage("Warning High")
+                        .build()
+        );
+        final org.slf4j.Logger logger = mock(
+                org.slf4j.LoggerFactory.getLogger(SLF4JAdapter.class)
+                        .getClass()
+        );
+        // act
+        report.print(new SLF4JAdapter(logger));
+        // assert
+        for (final Issue issue : report) {
+            verify(logger, never()).error(issue.toString());
+            verify(logger).warn(issue.toString());
+            verify(logger, never()).info(issue.toString());
+            verify(logger, never()).trace(issue.toString());
+        }
+    }
+
+    @Test
+    void shouldSLF4JLogWarningNormal() {
+        // arrange
+        final Report report = new Report();
+        report.add(
+                new IssueBuilder()
+                        .setSeverity(Severity.WARNING_NORMAL)
+                        .setMessage("Warning Normal")
+                        .build()
+        );
+        final org.slf4j.Logger logger = mock(
+                org.slf4j.LoggerFactory.getLogger(SLF4JAdapter.class)
+                        .getClass()
+        );
+        // act
+        report.print(new SLF4JAdapter(logger));
+        // assert
+        for (final Issue issue : report) {
+            verify(logger, never()).error(issue.toString());
+            verify(logger, never()).warn(issue.toString());
+            verify(logger).info(issue.toString());
+            verify(logger, never()).trace(issue.toString());
+        }
+    }
+
+    @Test
+    void shouldSLF4JLogWarningLow() {
+        // arrange
+        final Report report = new Report();
+        report.add(
+                new IssueBuilder()
+                        .setSeverity(Severity.WARNING_LOW)
+                        .setMessage("Warning Low")
+                        .build()
+        );
+        final org.slf4j.Logger logger = mock(
+                org.slf4j.LoggerFactory.getLogger(SLF4JAdapter.class)
+                        .getClass()
+        );
+        // act
+        report.print(new SLF4JAdapter(logger));
+        // assert
+        for (final Issue issue : report) {
+            verify(logger, never()).error(issue.toString());
+            verify(logger, never()).warn(issue.toString());
+            verify(logger, never()).info(issue.toString());
+            verify(logger).trace(issue.toString());
+        }
+    }
+
```

Comment:
Evtl. kann man die Zahl der Tests reduzieren, wenn man die vier Fälle in einem Test prüft.